pinc/stages.inc

```diff
@@ -11,360 +11,537 @@ include_once($relPath.'Pool.inc');
-        150000 => _('Flawless Formatter'),
-    ]
-);
-
-// -----------------------------------------------------------------------------
-
-new Round(
-    'F2',
-    _('Formatting Round 2'),
-    ['F1' => 400, 'days since reg' => 91], // 'F1' => 1000, 3 months after rollout
-    'REQ-HUMAN', // "peer approval"
-    _("Once you have met the requirements and requested access to F2, an evaluator will check over at least 150 of your recent F1 pages that have been completed in F2. If they've been formatted to F2 standards, you'll be granted access."),
-    null, // access_change_callback
-    _("The page-texts in this round need to be carefully checked to remove any remaining formatting errors."),
-    'formatting_guidelines.php',
-    $pi_tools_for_F,
-    null, // daily_page_limit
-    ['F1'],
-    [
-        0 => _('Refurbisher of Texts'),
-        25 => _('Sprucer of Texts'),
-        100 => _('Formatter Savant'),
-        500 => _('Formatting Wunderkind'),
-        1000 => _('Elite Formatter'),
-        2500 => _('Polisher of Texts'),
-        5000 => _('Formatting Artiste'),
-        10000 => _('Cultivator of Texts'),
-        20000 => _('Formatter of Enrichment'),
-        30000 => _('Designing Formatter'),
-        40000 => _('Formatting Artisan'),
-        50000 => _('Formatting Aficionado'),
-        60000 => _('Guru of Formatters'),
-        70000 => _('Formatting Familiar'),
-        80000 => _('Formatting Virtuoso'),
-        90000 => _('Formatter of Excellence'),
-        99000 => _('Exquisite Formatter'),
-        100000 => _('Formatting Specialist'),
-        110000 => _('Formatting Genius'),
-        120000 => _('Formatter of Fine Feats'),
-        130000 => _('Harmoniser of Texts'),
-        140000 => _('Formatting Architect'),
-        150000 => _('Preserver of Texts'),
-    ]
-);
-
-// ---------------------------
-
-// After creating all rounds:
-
-define('MAX_NUM_PAGE_EDITING_ROUNDS', 5);
-assert($n_rounds == MAX_NUM_PAGE_EDITING_ROUNDS);
-
-// ---------------------------
-
-declare_mentoring_pair('P1', 'P2');
-
-// -----------------------------------------------------------------------------
-
-// Some of the DP code assumes that the site has been configured with an
-// "Entry-Level Round" (ELR): a single round where new users start out and
-// become familiar with the site.  In particular, the page-tallies from this
-// round are distinguished from others in that they are (deemed to be)
-// a fair indicator of the user's experience.
-
-// This particular assignment assumes that the ELR is the first round
-// defined in this file. This is probably correct, but your site is
-// welcome to use a different round as your ELR.
-$ELR_round = get_Round_for_round_number(1);
-
-// On the other hand, if your site doesn't have a distinguished entry-level
-// round (e.g., everyone can work in multiple rounds from their first day),
-// then you may have to do some hacking (sorry).
-
+trait StatObj
```

Comment:
These definition and the access functions don't belong in `stages.inc` which should just be the creation of objects. They probably belong in their respective files with the traits in `Activity.inc`.

Reply:
I'll try going back to how the old code does it from inside the activity classes to see how it works out.